src/Psalm/Internal/Stubs/CoreGenericClasses.php

```diff
@@ -1151,3 +1151,21 @@ public function offsetGet ($object) {}
     public function getHash($object) {}
 
 }
+
+/**
+* @template T as object
+*
+* @property-read class-string<T> $name
+*/
+class ReflectionClass implements Reflector {
+
+    /**
+    * @var class-string<T>
+    */
+    public string $name;
```

Comment:
I don't think Psalm supports typed properties as of yet.

Reply:
see https://mobile.twitter.com/psalmphp/status/1083856311689928705

Reply:
Ah, great, somehow missed that.